tableau/virtual_connection_connections_data_source.go

```diff
@@ -0,0 +1,117 @@
+package tableau
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var (
+	_ datasource.DataSource              = &virtualConnectionConnectionsDataSource{}
+	_ datasource.DataSourceWithConfigure = &virtualConnectionConnectionsDataSource{}
+)
+
+func VirtualConnectionConnectionsDataSource() datasource.DataSource {
+	return &virtualConnectionConnectionsDataSource{}
+}
+
+type virtualConnectionConnectionsDataSource struct {
+	client *Client
+}
+
+type virtualConnectionConnectionsNestedDataModel struct {
+	ID            types.String `tfsdk:"id"`
+	DBClass       types.String `tfsdk:"db_class"`
+	ServerAddress types.String `tfsdk:"server_address"`
+	ServerPort    types.String `tfsdk:"server_port"`
+	UserName      types.String `tfsdk:"username"`
+}
+type virtualConnectionConnectionsDataSourceModel struct {
+	ID          types.String                                  `tfsdk:"id"`
+	Connections []virtualConnectionConnectionsNestedDataModel `tfsdk:"connections"`
+}
+
+func (d *virtualConnectionConnectionsDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_virtual_connection_connections"
+}
+
+func (d *virtualConnectionConnectionsDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Retrieve virtual connection's connections details",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:    true,
```

Comment:
should this be Required as the VirtualConnectionID to pass to the URL?

Reply:
Thats a good observation. I'll fix it.